bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBridgeHandler.java

```diff
@@ -0,0 +1,159 @@
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.touchwand.internal.discovery.TouchWandUnitDiscoveryService;
+import org.openhab.binding.touchwand.internal.dto.TouchWandUnitData;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TouchWandBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels TouchWand Wanderfull™ Hub channels .
+ *
+ * @author Roie Geron - Initial contribution
+ */
+@NonNullByDefault
+public class TouchWandBridgeHandler extends BaseBridgeHandler implements TouchWandUnitStatusUpdateListener {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BRIDGE);
+    private final Logger logger = LoggerFactory.getLogger(TouchWandBridgeHandler.class);
+    private final Map<ThingUID, @Nullable ServiceRegistration<?>> discoveryServiceRegs = Collections
+            .synchronizedMap(new HashMap<>());
+
+    private int statusRefreshRateSec;
+    private boolean addSecondaryUnits;
+    private BundleContext bundleContext;
+    private @Nullable TouchWandWebSockets touchWandWebSockets;
+    private Map<String, TouchWandUnitUpdateListener> unitUpdateListeners = new ConcurrentHashMap<>();
+
+    public TouchWandRestClient touchWandClient;
+
+    public TouchWandBridgeHandler(Bridge bridge, HttpClient httpClient, BundleContext bundleContext) {
+        super(bridge);
+        touchWandClient = new TouchWandRestClient(httpClient);
+        this.bundleContext = bundleContext;
+        touchWandWebSockets = null;
+    }
+
+    @Override
+    public void initialize() {
+        String host;
+        String port;
+        Configuration config;
+
+        updateStatus(ThingStatus.UNKNOWN);
+        config = getThing().getConfiguration();
+
+        host = config.get(HOST).toString();
+        port = config.get(PORT).toString();
+        statusRefreshRateSec = Integer.parseInt((config.get(STATUS_REFRESH_TIME).toString()));
+        addSecondaryUnits = Boolean.valueOf(config.get(ADD_SECONDARY_UNITS).toString());
+
+        scheduler.execute(() -> {
+            boolean thingReachable = false;
+            String password = config.get(PASS).toString();
+            String username = config.get(USER).toString();
+            thingReachable = touchWandClient.connect(username, password, host, port);
+            if (thingReachable) {
+                updateStatus(ThingStatus.ONLINE);
+                registerItemDiscoveryService(this);
+                touchWandWebSockets = new TouchWandWebSockets(host, scheduler);
+                touchWandWebSockets.registerListener(this);
+                touchWandWebSockets.connect();
```

Comment:
You could assign it to a local variable, too, to avoid the compiler warning.
```suggestion
                TouchWandWebSockets localSockets = touchWandWebSockets = new TouchWandWebSockets(host, scheduler);
                localSockets.registerListener(this);
                localSockets.connect();
```